### ISSUE TITLE
clock_getTime() with new Mac OS

### ISSUE THREAD

**Issue by ziazon** [wahern/cqueues#166]
cqueues fails to make on Mac OS. It throws errors about clock_getTime().

I fixed it by removing this chunk of code in: `../src/cqueues.c`

```
#if __APPLE__

#include <time.h>            /* struct timespec */

#include <errno.h>           /* errno EINVAL */

#include <sys/time.h>        /* TIMEVAL_TO_TIMESPEC struct timeval gettimeofday(3) */

#include <mach/mach_time.h>  /* mach_timebase_info_data_t mach_timebase_info() mach_absolute_time() */


#define CLOCK_REALTIME  0
#define CLOCK_VIRTUAL   1
#define CLOCK_PROF      2
#define CLOCK_MONOTONIC 3

static mach_timebase_info_data_t clock_timebase = {
    .numer = 1, .denom = 1,
}; /* clock_timebase */

void clock_gettime_init(void) __attribute__((constructor));

void clock_gettime_init(void) {
    if (mach_timebase_info(&clock_timebase) != KERN_SUCCESS)
        __builtin_abort();
} /* clock_gettime_init() */

static int clock_gettime(int clockid, struct timespec *ts) {
    switch (clockid) {
    case CLOCK_REALTIME: {
        struct timeval tv;

        if (0 != gettimeofday(&tv, 0))
            return -1;

        TIMEVAL_TO_TIMESPEC(&tv, ts);

        return 0;
    }
    case CLOCK_MONOTONIC: {
        unsigned long long abt;

        abt = mach_absolute_time();
        abt = abt * clock_timebase.numer / clock_timebase.denom;

        ts->tv_sec = abt / 1000000000UL;
        ts->tv_nsec = abt % 1000000000UL;

        return 0;
    }
    default:
        errno = EINVAL;

        return -1;
    } /* switch() */
} /* clock_gettime() */

#endif /* __APPLE__ */
```

**Comment by daurnimator** [comment]
From https://curl.haxx.se/mail/lib-2016-09/0049.html
> Apple did in fact declare the function with the appropriate availability, however they did so in a weakly linked manner. This means that even if you specify `-miphoneos-version-min=9.0` or `-mmacosx-version-min=10.11`, `clock_gettime` will still be declared at compile time. The only point that you can know if the function is available is by checking if `clock_gettime == null` at runtime.

The fix will probably look like this: https://github.com/civetweb/civetweb/pull/358/files

**Comment by wahern** [comment]
Can you try the latest HEAD?

**Comment by wahern** [comment]
This should be fixed with 1dcf2b8fa6f9eaef1879e2e45e82bc1afca3f1b1. Closing.
